test/libsolidity/semanticTests/storage/storage_boundary_struct_array_packed.sol

```diff
@@ -0,0 +1,125 @@
+contract C {
+    struct S {
+        uint64 a;
+        uint64 b;
+        uint64 c;
+        uint64 d;
+        // All fit in one slot (4 * 64 = 256 bits)
+    }
+
+    struct Neighbor {
```

Comment:
A more idiomatic name for this would be ["canary" or "sentinel"](https://en.wikipedia.org/wiki/Buffer_overflow_protection#Canaries). This would make it immediately clear what it's for.

```suggestion
    struct Canary {
```